x-pack/plugins/security_solution/public/management/pages/response_actions/view/response_actions_list.tsx

```diff
@@ -0,0 +1,422 @@
+  .euiText {
+    margin-top: 0.38rem;
+    overflow-y: visible !important;
+  }
+`;
+
+export const ResponseActionsList = memo<
+  EndpointActionListRequestQuery & { hideHeader?: boolean; hideHostNameColumn?: boolean }
+>(({ agentIds, commands, userIds, hideHeader = false, hideHostNameColumn = false }) => {
+  const getTestId = useTestIdGenerator('response-actions-list');
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<{
+    [k: ActionDetails['id']]: React.ReactNode;
+  }>({});
+
+  const [queryParams, setQueryParams] = useState<EndpointActionListRequestQuery>({
+    page: 0,
+    pageSize: 10,
+    agentIds,
+    commands,
+    userIds,
+  });
+
+  // initial fetch of list data
+  const {
+    error,
+    data: actionList,
+    isFetching,
+    isFetched,
+    refetch: reFetchEndpointActionList,
+  } = useGetEndpointActionList(queryParams, {
+    keepPreviousData: true,
+  });
+
+  // total actions
+  const totalItemCount = useMemo(() => actionList?.total, [actionList]);
+
+  // expanded tray contents
+  const toggleDetails = useCallback(
+    (item: ActionDetails) => {
+      const itemIdToExpandedRowMapValues = { ...itemIdToExpandedRowMap };
+      if (itemIdToExpandedRowMapValues[item.id]) {
+        delete itemIdToExpandedRowMapValues[item.id];
+      } else {
+        const {
+          startedAt,
+          completedAt,
+          isCompleted,
+          wasSuccessful,
+          isExpired,
+          command,
+          logEntries,
+        } = item;
+
+        // TODO: Remove this once we have
+        // ActionDetails.output, ActionDetails.parameters and ActionDetails.comments
+        const responseData = logEntries.reduce<
+          Array<LogsEndpointActionResponse['EndpointActions']['data']>
+        >((acc, curr) => {
+          if (curr.type === 'response') {
+            acc.push(curr.item.data.EndpointActions.data);
+          }
+          return acc;
+        }, []);
+
+        const descriptionListLeft = [
+          {
+            title: OUTPUT_MESSAGES.expandSection.placedAt,
+            description: `${startedAt}`,
+          },
+          {
+            title: OUTPUT_MESSAGES.expandSection.input,
+            description: `${command}`,
+          },
+          {
+            title: OUTPUT_MESSAGES.expandSection.output,
+            description: isExpired
+              ? OUTPUT_MESSAGES.hasExpired(command)
+              : isCompleted
+              ? wasSuccessful
+                ? OUTPUT_MESSAGES.wasSuccessful(command)
+                : OUTPUT_MESSAGES.hasFailed(command)
+              : OUTPUT_MESSAGES.isPending(command),
+          },
+        ];
+
+        const descriptionListCenter = [
+          {
+            title: OUTPUT_MESSAGES.expandSection.startedAt,
+            description: `${startedAt}`,
+          },
+          {
+            title: OUTPUT_MESSAGES.expandSection.parameters,
+            description: responseData.length ? responseData[0].parameters ?? '-' : '-',
```

Comment:
Use this function to get the security solution wide "empty value" for display:

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/x-pack/plugins/security_solution/public/common/components/empty_value/index.tsx#L20